### ISSUE TITLE
Make ffmpeg_build proxies script print more useful information

### ISSUE THREAD

**Issue by NathanLovato** [GDquest/Blender-power-sequencer#127]
- [x] Print transcoding settings
- [x] Print videos found and total number of jobs
- [x] videos remaining/done after each job
- [x] Remove commands print list
- [x] Make ffmpeg quieter -> just need to see the speed and progress

**Comment by sudopluto** [comment]
Hello @NathanLovato, I would like to tackle this issue, but I am relatively new to python and completely new to the project.

I feel confident in tackling this issue, but please allow me some time to get used to the codebase and 
look up parts of the program I don't understand.

I will try to report back here with my progress.

**Comment by NathanLovato** [comment]
Thanks much! I started refactoring the script in the `ffmpeg-build-script` branch. It's only one file for now: `scripts/ffmpeg_build_proxies.py`. Just let me finish #133 first, I'll do it today so you can get started ASAP!

If you have any questions please ask. I recommend to use the `print` function for the most part and maybe `logging` for extra info. It's a module dedicated to printing and debugging console programs:

```py
import logging
logging.debug('Debugging information')
logging.info('Informational message')
logging.warning('Warning:config file %s not found', 'server.conf')
logging.error('Error occurred')
logging.critical('Critical error -- shutting down')
```

Here's a quick tutorial on [when to use logging](https://docs.python.org/3/howto/logging.html)

**Comment by NathanLovato** [comment]
Okay done! check `scripts/ffmpeg_build_proxies.py`. Hopefully it's self-documented. You'll find the script at the end of the file.
I would recommend:

1. Please use the `print` function only in the script at the end (the actual script is in the last ~10 lines of code)
2. If you feel like there's useful debug info to print in functions, you should use `logging` instead, so that the end user doesn't have to bother about that

Note I don't know how to best `Print transcoding settings`. I'm sure there's some configuration on ffmpeg to do that but I haven't had time to look into it. Same for printing progress and ffmpeg jobs without flooding the command line. You can use the print's `flush` argument to update the print in place if I recall correctly but I'm not sure how to best do that!

**Comment by NathanLovato** [comment]
If you have questions or need help ask anytime :)

**Comment by sudopluto** [comment]
Cool, will get started on this soon.

**Comment by sudopluto** [comment]
just a quick progress update: i got the output to look like this:

```
pranav@pranav-notebook:~/git/Blender-power-sequencer/scripts$ python build_ffmpeg_proxies.py ~/test/
found 3 file(s) to convert
    /home/pranav/test/big_buck_bunny.mp4
    /home/pranav/test/big_buck_bunny (copy).mp4
    /home/pranav/test/big_buck_bunny (another copy).mp4
1/3 :: /home/pranav/test/big_buck_bunny.mp4
python>>>frame= 1099 fps=0.0 q=2.0 size=    1865kB time=00:00:45.66 bitrate= 334.6kbits/s speed=91.2x    

python>>>frame= 1440 fps=0.0 q=2.0 Lsize=    2474kB time=00:00:59.91 bitrate= 338.3kbits/s speed=92.3x    

2/3 :: /home/pranav/test/big_buck_bunny (copy).mp4
python>>>frame=  852 fps=0.0 q=2.0 size=    1482kB time=00:00:35.37 bitrate= 343.3kbits/s speed=70.7x    

python>>>frame= 1440 fps=0.0 q=2.0 Lsize=    2474kB time=00:00:59.91 bitrate= 338.3kbits/s speed=62.1x    

3/3 :: /home/pranav/test/big_buck_bunny (another copy).mp4
python>>>frame=  804 fps=0.0 q=2.0 size=    1390kB time=00:00:33.37 bitrate= 341.2kbits/s speed=66.7x    

python>>>frame= 1440 fps=0.0 q=2.0 Lsize=    2474kB time=00:00:59.91 bitrate= 338.3kbits/s speed=  77x
```
the explanation behind the python>>>, is that means the ffmpeg info is being printed from pythons side
of things. soon I will parse that output, and use it provide a percentage based progress indicator

could you please explain the "remove commands print list" and the transcoding settings in more detail?

also please take a look at the code so far: https://github.com/sudopluto/Blender-power-sequencer/commit/6a54a201c9183991e25b6439e9f2dc5e6491aef6#diff-2e758528bf46eace2038efbbe1f6aef6

**Comment by NathanLovato** [comment]
Looking good! Thanks :)
`remove commands print list` is done already.
You can forget about `transcoding settings`. It'll be for when we add support to 25%, 50% proxies (and maybe quality options).

**Comment by sudopluto** [comment]
ok, so I have the current output looking like this:
![cool!](https://i.imgur.com/zN84MnX.gif)

while i believe the parsing is robust, it could benefit from extra testing.

the code in general could use some clean up. I feel that one exploration could be to add
some classes, as there are many if statements now for deciding behaviour based on if the 
file is an image or not

also, for some reason, image proxies are being generated multiple times, and aren't ignored like
videos are, going to log a second bug report for this

**Comment by NathanLovato** [comment]
Looking great, thanks much!

> I feel that one exploration could be to add
> some classes, as there are many if statements now for deciding behaviour based on if the
> file is an image or not

Definitely. Especially to add some settings and interface it with Blender later. Like a lot of things in the add-on, it started as quick and dirty code to edit faster (or in this case reduce the proxies' size). I just have little time to work on the add-on overall so dunno when I could get to it. 

Would you like to do this? Your help would be much welcome!

**Comment by NathanLovato** [comment]
Just fixed the other bug in 8cef8ab986dcf4d9f03b9f08ac88fa48699c8cfc. Can you pull and rebase? I renamed `proxy_folder_path` to `proxy_file_path` as the variable contains the full file path - it caused another bug

**Comment by NathanLovato** [comment]
Big thanks for the PR :)
